journal/src/test/java/io/camunda/zeebe/journal/file/SegmentedJournalTest.java

```diff
@@ -361,6 +365,72 @@ void shouldUpdateIndexMappingsAfterRestart() {
         .isEqualTo(indexBeforeClose.lookup(secondIndexedPosition).position());
   }
 
+  @Test
```

Comment:
NIT: Since this class is becoming bigger, we can also move all corruption tests to a separate test class.